spec/tree_spec.rb

```diff
@@ -2,13 +2,93 @@
 require_relative '../lib/tree'
 
 describe Tree do
+  let(:tree) { Tree.new(1, [], true, 1) }
```

Comment:
nice use of `.let`. in rails apps, you'll often see use of a library like factory_bot for building factories to use in test. there are also fixture libraries that are popular.

https://github.com/thoughtbot/factory_bot